### ISSUE TITLE
Next's resolve.modules webpack config makes it impossible to use some dependencies

### ISSUE THREAD

**Issue by exogen** [zeit/next.js#5164]
# Bug report

## Describe the bug

Due to `NEXT_PROJECT_ROOT_NODE_MODULES` coming first in Next's `resolve.modules` webpack setting, I'm unable to use some dependencies in my own code.

For example, I'm trying to use [debug](https://www.npmjs.com/package/debug) v4, which has a new `.extend()` API.

Even though I'm declaring and using the dependency correctly in my own package, the version of `debug` that appears in Next's `node_modules` folder (due to its direct dependency on `send`, which uses `debug`) takes precedence, and thus my code breaks, because it's an older version (2.6.9):

```
└─┬ next@6.1.2
  ├─┬ @babel/core@7.0.0-beta.42
  │ ├─┬ @babel/traverse@7.0.0-beta.42
  │ │ └── debug@3.2.5
  │ └── debug@3.2.5
  ├─┬ send@0.16.1
  │ └── debug@2.6.9
  └─┬ write-file-webpack-plugin@4.2.0
    └── debug@2.6.9
```

Putting `NEXT_PROJECT_ROOT_NODE_MODULES` first here seems very dangerous, because not only Next's direct `dependencies` will appear there, but due to the way npm flattening/hoisting works, any transitive dependency could appear there too.

It seems like it would be much safer to just alias any specific known dependencies that absolutely must come from `next/node_modules`.

## To Reproduce

1. Depend on `debug` 4.0.1.
2. Use the `require('debug')('module').extend('submodule')` feature in a page.
3. See error:

```
next-dev.js:57 __WEBPACK_IMPORTED_MODULE_3_debug___default(...)(...).extend is not a function
TypeError: __WEBPACK_IMPORTED_MODULE_3_debug___default(...)(...).extend is not a function
    at Object../components/IntlProvider.js (https://localhost:3000/_next/-/page/_app.js:3978:66)
    at __webpack_require__ (https://localhost:3000/_next/static/commons/manifest.js:715:31)
    at fn (https://localhost:3000/_next/static/commons/manifest.js:118:20)
    at Object.<anonymous> (https://localhost:3000/_next/-/page/_app.js:12045:83)
    at Object../pages/_app.js (https://localhost:3000/_next/-/page/_app.js:12193:30)
    at __webpack_require__ (https://localhost:3000/_next/static/commons/manifest.js:715:31)
    at fn (https://localhost:3000/_next/static/commons/manifest.js:118:20)
    at Object.1 (https://localhost:3000/_next/-/page/_app.js:12357:18)
    at __webpack_require__ (https://localhost:3000/_next/static/commons/manifest.js:715:31)
    at webpackJsonpCallback (https://localhost:3000/_next/static/commons/manifest.js:26:23)
```

## Expected behavior

I can use the version of `debug` that I declared in my `package.json`.

## System information
 - OS: macOS
 - 6.1.2, canary

**Comment by timneutkens** [comment]
I believe there was a reason for this ordering, something with users installing packages that got overwritten inside Next.js too or something.

**Comment by exogen** [comment]
Is there a list of packages that absolutely must be resolved from Next's `node_modules`? I'm curious why webpack's normal dependency resolution wouldn't just do the right thing – if it sees an import coming from somewhere in the `next` package, the resolution shouldn't be a problem there.

So if this `resolve.modules` setting is necessary, that means there's likely something in user-defined code (`pages` or `next.config.js`?) that must come from Next.js' dependencies. Nothing jumps to mind though, as users should really only be importing from `react`, `react-dim`, `next`, and then whatever modules they've installed in their own `dependencies`...

What breaks if it's removed from `resolve.modules`? I might play around with it, as its annoying to have to alias certain dependencies only have discovering that they break.

**Comment by timneutkens** [comment]
Fixed in #6133
